nautobot/dcim/filters/__init__.py

```diff
@@ -1116,26 +1122,39 @@ class Meta:
             "interface_redundancy_groups",
         ]
 
+    def generate_query_filter_device(self, value):
+        if not hasattr(value, "__iter__") or isinstance(value, str):
+            value = [value]
+
+        device_ids = set(str(item) for item in value if is_uuid(item))
+        device_names = set(str(item) for item in value if not is_uuid(item))
+        devices = Device.objects.filter(Q(name__in=device_names) | Q(pk__in=device_ids))
+        all_interface_ids = []
+        for device in devices:
+            all_interface_ids.extend(device.vc_interfaces.values_list("id", flat=True))
+        return Q(pk__in=all_interface_ids)
```

Comment:
Does this keep the old behavior where if no `devices` are found (because we specified a nonexistent PK or name), the query fails closed?

Reply:
No, but it should. I'll fix that.

Reply:
This is actually handled by the `Filter.field` in `NaturalKeyOrPKMultipleChoiceFilter` so we don't need to do it in the method.